nov.el

```diff
@@ -797,6 +799,38 @@ Saving is only done if `nov-save-place-file' is set."
 (add-hook 'nov-mode-hook 'nov-add-to-recentf)
 (add-hook 'nov-mode-hook 'hack-dir-local-variables-non-file-buffer)
 
+
+(defun nov--find-file (file index position)
+  "Open FILE(nil means current buffer) in nov-mode and go to the specified INDEX and POSITION."
+  (when file
+    (find-file file))
+  (unless (eq major-mode 'nov-mode)
+    (nov-mode))
+  (when (not (nov--index-valid-p nov-documents index))
+    (error "Invalid documents index"))
+  (setq nov-documents-index index)
+  (nov-render-document)
+  (goto-char point))
+
+;; Bookmark Integration
+(defun nov-bookmark-make-record  ()
+  "Create a bookmark epub record."
+  (cons (buffer-name)
+        `((filename . ,nov-file-name)
+          (index . ,nov-documents-index)
+          (position . ,(point))
+          (handler . nov-bookmark-jump-handler))))
+
+;;;###autoload
```

Comment:
Why the autoload? Does following a bookmark not work otherwise?

Reply:
I will remove it